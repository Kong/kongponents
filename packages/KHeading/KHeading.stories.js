import { storiesOf } from '@storybook/vue'
import { default as KHeading, sizes } from './KHeading.vue'

const headings = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6']

storiesOf('Heading', module)
  .add('Default', () => ({
    components: { KHeading },
    template: `<KHeading>Default Heading</KHeader>`
  }), {info: {}})
  .add('Headings', () => ({
    components: { KHeading },
    template: `
    <div>
      ${headings.map(type => {
      return `<KHeading style="margin: 0 0 1rem" type="${type}">Heading ${type}</KHeading>`
    }).join('')}
    </div>`
  }), {info: {}})
  .add('Sizes', () => ({
    components: { KHeading },
    template: `
    <div>
      ${Object.values(sizes).map(size => {
      return `<KHeading type="p" size="${size}">${size}</KHeading>`
    }).join('')}
    </div>`
  }), {info: {}})
  .add('Fonts', () => ({
    components: { KHeading },
    template: `
    <div>
      <div>
        <h1>Sans Fonts</h1>
        <div>${Object.values(sizes).map(size => {
      return `<KHeading style="margin: 0 0 1rem" type="p" size="${size}" font="sans">${size}</KHeading>`
    }).join('')}</div>
      </div>
      <div>
        <h1>Mono Fonts</h1>
        <div>${Object.values(sizes).map(size => {
      return `<KHeading style="margin: 0 0 1rem" type="p" size="${size}" font="mono">${size}</KHeading>`
    }).join('')}</div>
      </div>
    </div>
    `
  }))
