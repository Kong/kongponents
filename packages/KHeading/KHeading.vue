<template>
  <component
    :is="type"
    :class="[size, font]"
    class="k-heading">
    <slot/>
  </component>
</template>

<script>
export const sizes = {
  xxxl: 'xxxl',
  xxl: 'xxl',
  xl: 'xl',
  lg: 'lg',
  md: 'md',
  sm: 'sm',
  xs: 'xs'
}

export default {
  name: 'KHeading',
  props: {
    /**
     * Sets the element type, if not set defaults to h1
     */
    type: {
      type: String,
      default: 'h1'
    },

    /**
     * Sets size of heading
     */
    size: {
      type: String,
      default: 'xxxl',
      validator: (value) => Object.values(sizes)
        .indexOf(value) !== -1
    },

    /**
     * Sets font type, defaults as sans
     */
    font: {
      type: String,
      default: 'sans',
      validator: (value) => ['sans', 'mono']
        .indexOf(value) !== -1
    }
  }
}
</script>

<style scoped>
</style>
