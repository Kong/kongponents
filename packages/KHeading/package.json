{
  "name": "@kongponents/kheading",
  "version": "0.1.0",
  "description": "KHeading description here.",
  "main": "KHeading.vue",
  "scripts": {
    "test": "jest"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Kong/kongponents.git"
  },
  "author": "Kong Inc",
  "license": "ISC",
  "bugs": {
    "url": "https://github.com/Kong/kongponents/issues"
  },
  "homepage": "https://github.com/Kong/kongponents#readme"
}
