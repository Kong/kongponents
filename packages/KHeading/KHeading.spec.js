import { mount } from '@vue/test-utils'
import KHeading from '@/KHeading/KHeading'

describe('KHeading', () => {
  it('renders as a span', () => {
    const wrapper = mount(KHeading, {
      propsData: {
        type: 'span'
      }
    })

    expect(wrapper.is('span')).toBe(true)
  })

  it('renders as an h4', () => {
    const wrapper = mount(KHeading, {
      propsData: {
        type: 'h4'
      }
    })

    expect(wrapper.is('h4')).toBe(true)
  })

  it('renders with md & mono class', () => {
    const wrapper = mount(KHeading, {
      propsData: {
        size: 'md',
        font: 'mono'
      }
    })

    expect(wrapper.classes()).toContain('md', 'mono')
  })

  it('matches snapshot', () => {
    const wrapper = mount(KHeading)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
