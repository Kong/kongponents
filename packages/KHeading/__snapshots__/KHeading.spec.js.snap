// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KHeading matches snapshot 1`] = `<h1 class="k-heading xxxl sans"></h1>`;
